### DISCUSSION TITLE
Issue using calibrated camera and Viewer class

### DISCUSSION THREAD

**Discussion by thesvend** [libigl/libigl#2274] [General]
#### Describe your issue
Hello, I am trying to use a "virtual camera" to render an image of my pointcloud. What I have done so far is that I used a 3D camera giving me a pointcloud and an RGB image. I imported the pointcloud into the igl::viewer and set all camera parameters to the known values from the camera. When I compare the position of the points in the RGB Image with the one in my igl window there seems to be some sort of scaling that I don't know how to get rid of. I don't know if thats an actual issue or just me misunderstanding the usage of these parameters.

Here are 2 images that show the issue (first: 3D-camera   second: igl-viewer)
![img1](https://user-images.githubusercontent.com/45479696/57936668-a80f2a00-78c4-11e9-934d-9ae39a69174a.png) 
![img_libigl](https://user-images.githubusercontent.com/45479696/57936754-cd9c3380-78c4-11e9-967a-70864e5f3996.PNG)

The code that I used to set the viewer parameters:
```
viewer.core.camera_center << 0, 0, 0;       // cameras origin
viewer.core.camera_translation << 0, 0, 0;  // also cameras origin?
viewer.core.camera_eye << 0, 0, -1;         // camera view direction
viewer.core.camera_up << 0, -1, 0;          // camera up
viewer.core.camera_view_angle = 42.27;      // FOV in vertical direction
viewer.resize(1920, 1200);                  // Camera resolution
```

It would be great to get some advice on how to fix this. thank you :)
#### Check all that apply (change to `[x]`)
- [x] Windows
- [ ] Mac OS X
- [ ] Linux
- [ ] I have tried the `dev` branch and the problem persists

**Comment by jdumas** [comment]
Hi. Camera settings are subject to change. They are already a bit different in the `dev` branch, and yes they need better documentation (in the meantime you can checkout related issues such as #1076). To answer your specific question: probably your issue may be due to the fact that there is a call to [align_camera_center()](https://github.com/libigl/libigl/blob/10e04cbef5bc54b78439919046781a6e8d8cde1e/include/igl/opengl/glfw/Viewer.cpp#L214) when the viewer is initialized.

But honestly I would also suggest to use a 2D quad with a texture showing your captured image in the same scene as your reconstructed 3D point cloud, so you don't get "alignment issues" to control of the libigl viewer camera.

**Comment by thesvend** [comment]
Hi Jérémie, thanks for the reply. Nice to hear that someone is already working on these topics. I will take a deeper look into the align_camera_center() function and try to figure out what happens inside of it. 

With a "2D quad with texture" you mean a colored quad-mesh with the faces having the size of my cameras pixels? Where do you suggest to place that plane in terms of distance from the cameras center? As far as I understood your workaround: I can simpl call the align_camer_center() once I have the plane in the same viewer as my pointcloud.. Then the window should automatically resize to that correct size. Is that correct?

**Comment by jdumas** [comment]
By quad mesh I mean you draw a quad (i.e. 2 triangles) with a texture (see tutorial 607_ScreenCapture), and you set as texture the image captured by your camera. Then by applying the inverse of your camera matrix onto the quad vertices positions, you should be able to place the quad in world-space in a way that aligns with your point cloud (at least that's a rough idea).

**Comment by thesvend** [comment]
Ah okay, now I get what you mean. What would I use as initial positions for the quad vertices before I apply the inverse of the camera matrix? I know that basically everything that defines a square would work to make the captured image look like an image int he viewer but I am unsure on what vertex positions I should use to also get a correct scaling. In the tutorial a simple 'unit square' is used.
